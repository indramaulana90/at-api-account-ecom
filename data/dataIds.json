[
	{
		"relativePath": "__tests__/v1/example/post/example/data.js",
		"ids": {
			"idUsers": 999999999
		}
	},
	{
		"relativePath": "__tests__/v1/order/get/order_parcel_detail/data.js",
		"ids": {
			"idOrders": 839640065,
			"idOrderParcels": 839641572,
			"idOrderParcelProducts": 839644779,
			"idBrands": "888888f39970108f39970108",
			"idBuyers": 839966387,
			"idUrbans": 839966387,
			"idInvoiceGroups": "888888f39970108f39970108",
			"idCart": "888888f39970314f39970314",
			"idDevice": "839966387",
			"idProducts": "888888f39970108f39970108",
			"idPhoneNumber": 839981005
		}
	},
	{
		"relativePath": "__tests__/v1/salesman_kpi/get/workday_list/data.js",
		"ids": {
			"idWorkingDays": [
				841736409,
				841736511,
				841736535
			]
		}
	},
	{
		"relativePath": "__tests__/v1/salesman_kpi/get/workday_detail/data.js",
		"ids": {
			"idWorkingDays": [
				841802581,
				841802625
			]
		}
	},
	{
		"relativePath": "__tests__/v1/salesman_kpi/delete/workingDay/data.js",
		"ids": {
			"idWorkingDays": [
				841888623,
				841888635,
				841888672
			]
		}
	},
	{
		"relativePath": "__tests__/v1/salesman_kpi/patch/update_workingday/data.js",
		"ids": {
			"idWorkingDays": [
				841909018,
				841909045,
				841909072
			]
		}
	},
	{
		"relativePath": "__tests__/v1/order/post/cancel_order_parcel/data.js",
		"ids": {
			"idOrders": 842063620,
			"idOrderParcels": 842063620,
			"idOrderParcelProducts": 842063620,
			"idBrands": "888888f42063620f42063620",
			"idBuyers": 842063620,
			"idUrbans": 842063620,
			"idInvoiceGroups": "888888f42063620f42063620",
			"idCart": "888888f42063620f42063620",
			"idDevice": 842063620,
			"idProducts": "888888f42063620f42063620",
			"idPhoneNumber": 842063620
		}
	},
	{
		"relativePath": "__tests__/v1/salesman_kpi/get/holiday_list/data.js",
		"ids": {
			"idHolidays": [
				841986121,
				841986145,
				841986166,
				841986181,
				841986198
			]
		}
	},
	{
		"relativePath": "__tests__/v1/salesman_kpi/get/holiday_detail/data.js",
		"ids": {
			"idHolidays": [
				841994725,
				841994752
			]
		}
	},
	{
		"relativePath": "__tests__/v1/salesman_kpi/delete/holidays/data.js",
		"ids": {
			"idHolidays": [
				841999606,
				841999633,
				841999652
			]
		}
	},
	{
		"relativePath": "__tests__/v1/salesman_kpi/post/holiday/data.js",
		"ids": {
			"idHolidays": 842147917
		}
	},
	{
		"relativePath": "__tests__/v1/salesman_kpi/patch/holiday_update/data.js",
		"ids": {
			"idHolidays": [
				842318652,
				842318674,
				842318692,
				842321145
			]
		}
	},
	{
		"relativePath": "__tests__/v1/order/patch/update_order_status/data.js",
		"ids": {
			"idOrders": 842412325,
			"idOrderParcels": 842412325,
			"idOrderParcelProducts": 842412325,
			"idBrands": "888888f42412325f42412325",
			"idBuyers": 842412325,
			"idUrbans": 842412325,
			"idInvoiceGroups": "888888f42412325f42412325",
			"idCart": "888888f42412325f42412325",
			"idDevice": 842412325,
			"idProducts": "888888f42412325f42412325",
			"idPhoneNumber": 842412325
		}
	},
	{
		"relativePath": "__tests__/v1/buyer_order/patch/order_sellers_is_read/data.js",
		"ids": {
			"idOrders": 856578262,
			"idOrderSellers": 856578315
		}
	},
	{
		"relativePath": "__tests__/v1/buyer_order/patch/update_order_seller_failed/data.js",
		"ids": {
			"idOrders": 856642262,
			"idOrderSellers": 856642262,
			"idCart": "888888f56642550f56642550",
			"idDevice": 856642550,
			"idOrderBilling": 856642736,
			"idOrderSellerStatusHistoriesId": 856642736,
			"idOrderProduct": 856642736
		}
	},
	{
		"relativePath": "__tests__/v1/buyer_order/patch/update_order_seller_cancel/data.js",
		"ids": {
			"idOrders": 856664266,
			"idOrderSellers": 856664266
		}
	},
	{
		"relativePath": "__tests__/v1/warehouse/post/assign_product/data.js",
		"ids": {
			"idWarehouses": 862452416,
			"idSyncProduct": "62ec8be49c912a4015db7b41"
		}
	},
	{
		"relativePath": "__tests__/v1/warehouse/patch/warehouse_assign_product/data.js",
		"ids": {
			"idSyncProduct": "62ec8be49c912a4015db7b41"
		}
	}
]