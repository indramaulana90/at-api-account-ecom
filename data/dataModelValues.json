{
    "banners":{
        "name": "ATAPI ${params.atName} Banner Name",
        "header": "ATAPI${params.atName} Header",
        "description": "ATAPI ${params.atName} Description",
        "termAndCondition": "<p>• Berlaku untuk user dari semua tier</p><p>• Pembelian min 1.000.000 untuk SKU (SGM 300gr)</p><p>• Tidak berlaku kelipatan</p>"
    },
    "brands": {
        "name": "ATAPI ${params.atName} Brand Name",
        "code": "ATAPI${params.atName}Brand"
    },
    "buyers": {
        "name": "ATAPI ${params.atName} Buyers Name",
        "address": "ATAPI ${params.atName} Address"
    },
    "catalogues": {
        "name": "ATAPI ${params.atName} Catalogue Name",
        "code": "ATAPI${params.atName}Cata",
        "detail": "ATAPI ${params.atName} Catalogue Detail",
        "description": "ATAPI ${params.atName} Catalogue Description",
        "information": "ATAPI ${params.atName} Catalogue Information"
    },
    "catalogueCategories": {
        "name": "ATAPI ${params.atName} Category Name",
        "code": "ATAPI${params.atName}CCat"
    },
    "catalogueUnits": {
        "name": "ATAPI ${params.atName} Catalogue Unit Name"
    },
    "channels": {
        "name": "ATAPI ${params.atName} Channel Name",
        "externalId": "ATAPI${params.atName}Chan",
        "description": "ATAPI ${params.atName} Description"
    },
    "clusters": {
        "name": "ATAPI ${params.atName} Cluster Name",
        "externalId": "ATAPI${params.atName}Clus",
        "description": "ATAPI ${params.atName} Description"
    },
    "groups": {
        "name": "ATAPI ${params.atName} Group Name",
        "externalId": "ATAPI${params.atName}Grup",
        "description": "ATAPI ${params.atName} Description"
    },
    "invoiceGroups": {
        "name": "ATAPI ${params.atName} Invoice Group name",
        "code": "ATAPI${params.atName}InGr",
        "externalId": "ATAPI${params.atName}InGr"
    },
    "mssCatalogues": {
        "name": "ATAPI ${params.atName} MssCatalogues Name",
        "externalId": "ATAPI${params.atName}MSS"
    },
    "portfolios": {
        "name": "ATAPI ${params.atName} Portfolio Name",
        "externalId": "ATAPI${params.atName}Ptfl"
    },
    "products": {
        "name": "ATAPI ${params.atName} Product Name",
        "externalId": "ATAPI${params.atName}Pdct"
    },
    "stores": {
        "address": "ATAPI ${params.atName} Stores Address",
        "name": "ATAPI ${params.atName} Stores Name",
        "noteAddress": "ATAPI ${params.atName} Stores Note Address",
        "externalId": "ATAPI${params.atName}Strs"
    },
    "salesRep": {
        "name": "ATAPI ${params.atName} Users Sales fullname",
        "externalId": "ATAPI${params.atName}UsSl"
    },
    "salesTeamStoreChannels": {
        "name": "ATAPI ${params.atName} ST Store Channels Name"
    },
    "types": {
        "name": "ATAPI ${params.atName} Type Name",
        "externalId": "ATAPI${params.atName}Type",
        "description": "ATAPI ${params.atName} Description"
    },
    "urbans": {
        "city": "ATAPI ${params.atName} City",
        "district": "ATAPI ${params.atName} District",
        "urban": "ATAPI ${params.atName} Urban"
    },
    "users": {
        "fullname": "ATAPI ${params.atName} users fullname",
        "email":"ATAPI${params.atName}@atapi.com",
        "userCode": "ATAPI${params.atName}Usrs"
    },
    "usersSales": {
        "fullname": "ATAPI ${params.atName} users sales fullname",
        "email":"ATAPI${params.atName}@sales.com",
        "userCode": "ATAPI${params.atName}UsSl"
    },
    "vouchers": {
        "name": "ATAPI {params.atName} Name",
        "externalId": "ATAPI${params.atName}Vouc",
        "code": "${params.atName}",
        "description": "ATAPI ${params.atName} Description",
        "header":"ATAPI ${params.atName} Header",
        "terms": "{ATAPI ${params.atName} Terms and Condition}",
        "instructions": "{ATAPI ${params.atName} Instructions}",
        "tag": "{ATAPI ${params.atName} Tag}"
    },
    "warehouses": {
        "name": "ATAPI ${params.atName} Warehouse name",
        "code": "ATAPI${params.atName}WH",
        "externalId": "ATAPI${params.atName}WH",
        "noteAddress": "ATAPI ${params.atName} Note Address",
        "address": "ATAPI ${params.atName} Address"
    },
    "warehouseValues": {
        "name": "ATAPI ${params.atName} Warehouse Value name"
    },
    "warehouseTemperatures": {
        "name": "ATAPI ${params.atName} Warehouse Temperature name"
    }
}