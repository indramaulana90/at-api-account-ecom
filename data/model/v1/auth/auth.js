/**
* Author      : yosua siagian (yosua@sinbad.co.id)
* Modifier    :
* Domain      : auth
* Description : This File for Model Seeding Data on auth table
*/

import seedingData from '@at-api-core/seeding-data';

/**
 * This function for seeding data select by id on auth table.
 * @param {Number} idUser auth user id.
 * @return {ArrayOfObject} result from db.
 */
export const selectByIdAuth = async (idUser) => {
  const query = {
    type: 'select',
    table: 'public.auth',
    condition: { idUser },
    limit: 5,
  };

  const result = await seedingData(query, 'service_auth');

  return result;
};

/**
 * This function for seeding data delete by id on auth table.
 * @param {Object} params object on will be inserted with:
 * @param {Number} params.mobilePhoneAuth mobile phone from user table.
 */
export const deleteAuthByNumber = async (params) => {
  const query = {
    type: 'remove',
    table: 'public.auth',
    condition: {
      mobile_phone: params.mobilePhoneAuth,
    },
  };

  await seedingData(query, 'service_auth');
};
