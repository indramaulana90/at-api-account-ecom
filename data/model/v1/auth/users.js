/**
* Author      : Evinna Ginola Utama (evinna.ginola@sinbad.co.id)
* Modifier    :
* Domain      : auth
* Description : This File for Model Seeding Data on users table
*/

import seedingData from '@at-api-core/seeding-data';

/**
 * This function for seeding data select by id on users table.
 * @param {Number} id users id.
 * @return {ArrayOfObject} result from db.
 */
export const selectById = async (id) => {
  const query = {
    type: 'select',
    table: 'public.users',
    condition: { id },
  };

  const result = await seedingData(query, 'service_auth');

  return result;
};

/**
 * This function for seeding data select by id on users table.
 * @param {Number} phone users mobile_phone.
 * @return {ArrayOfObject} result from db.
 */
export const selectByPhone = async (phone) => {
  const query = {
    type: 'select',
    table: 'public.users',
    condition: { mobile_phone: phone },
  };

  const result = await seedingData(query, 'service_auth');

  return result;
};

/**
 * This function for seeding data reset account by id on users table.
 * @param {Number} id users id.
 */
export const resetAccountChimera = async (id) => {
  const query = {
    type: 'update',
    table: 'public.users',
    modifier: {
      name: 'Chimera-SR',
      image_url: 'https://sinbad.co.id/image.jpg',
    },
    condition: { id },
  };

  await seedingData(query, 'service_auth');
};

/**
 * This function for seeding data select by id on users table.
 * @param {Number} id users id.
 */
export const deleteById = async (id) => {
  const query = {
    type: 'remove',
    table: 'public.users',
    condition: { id },
  };

  await seedingData(query, 'service_auth');
};
