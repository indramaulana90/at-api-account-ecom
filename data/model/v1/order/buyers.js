/**
* Author      : Farhad (farhad@sinbad.co.id)
* Modifier    :
* Domain      : order
* Description : This File for Model Seeding Data on buyers table
*/

import seedingData from '@at-api-core/seeding-data';
import getModelValues from '@at-api-core/get-model-values';

const dateCommander = require('@at-api-core/date-commander');

/**
 * This function for seeding data select by id on buyers table.
 * @param {Number} id buyers id.
 * @return {ArrayOfObject} result from db.
 */
export const selectById = async (id) => {
  const query = {
    type: 'select',
    table: 'public.buyers',
    condition: { id },
    limit: 5,
  };

  const result = await seedingData(query, 'service_order');

  return result;
};

/**
 * This function for seeding data insert with half params on buyers table.
 * @param {Object} params object on will be inserted with:
 * @param {Number} params.idBuyers id from buyers table.
 * @param {String} params.idUrbans id from urbans table.
 */
export const insertHalfParams = async (params) => {
  const query = {
    type: 'insert',
    table: 'public.buyers',
    values: {
      id: params.idBuyers,
      name: getModelValues('buyers', params).name,
      address: getModelValues('buyers', params).address,
      urban_id: params.idUrbans,
      tax_no: '',
      created_at: dateCommander('now'),
      updated_at: dateCommander('now'),
      deleted_at: null,
    },
  };

  await seedingData(query, 'service_order');
};

/**
 * This function for seeding data delete by id on buyers table.
 * @param {Object} params object on will be inserted with:
 * @param {Number} params.idBuyers id from buyers table.
 */
export const deleteById = async (params) => {
  const query = {
    type: 'remove',
    table: 'public.buyers',
    condition: {
      id: params.idBuyers,
    },
  };

  await seedingData(query, 'service_order');
};
