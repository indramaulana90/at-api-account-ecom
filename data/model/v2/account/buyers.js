/**
* Author      : Immanuel Yohansen (immanuel.yohansen@sinbad.co.id)
* Modifier    :
* Domain      : account
* Description : This File for Model Seeding Data on buyers table
*/

import seedingData from '@at-api-core/seeding-data';

/**
 * This function for seeding data select by id on buyers table.
 * @param {Number} id buyers id.
 * @return {ArrayOfObject} result from db.
 */
export const selectById = async (id) => {
  const query = {
    type: 'select',
    table: 'public.buyers',
    condition: { id },
    limit: 5,
  };

  const result = await seedingData(query, 'ms_ng_ecom_account');

  return result;
};

/**
 * This function for seeding data insert with half params on buyers table.
 * @param {Object} params object on will be inserted with:
 * @param {Number} params.idBuyers id from buyers table.
 * @param {String} params.atName Original name of AT Name without space maximum 11 char (e.g. StoreDtail).
 */
export const updateLocationId = async (id, locationId) => {
  const query = {
    type: 'update',
    table: 'public.buyers',
    modifier: { location_id: locationId },
    condition: { id },
  };
  const result = await seedingData(query, 'ms_ng_ecom_account');
  return result;
};
