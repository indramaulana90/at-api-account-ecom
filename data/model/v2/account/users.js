/**
* Author      : Immanuel Yohansen (immanuel.yohansen@sinbad.co.id)
* Modifier    :
* Domain      : account
* Description : This File for Model Seeding Data on users table
*/

import seedingData from '@at-api-core/seeding-data';

/**
 * This function for seeding data select by name on users table.
 * @param {Number} name users name.
 * @return {ArrayOfObject} result from db.
 */
export const selectByName = async (name) => {
  const query = {
    type: 'select',
    table: 'public.users',
    condition: { name },
    limit: 5,
  };

  const result = await seedingData(query, 'ms_ng_ecom_account');

  return result;
};
