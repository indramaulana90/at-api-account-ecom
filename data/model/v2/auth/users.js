/**
* Author      : Immanuel Yohansen (immanuel.yohansen@sinbad.co.id)
* Modifier    :
* Domain      : auth
* Description : This File for Model Seeding Data on users table
*/

import seedingData from '@at-api-core/seeding-data';

/**
 * This function for seeding data select by id on users table.
 * @param {Number} id users id.
 * @return {ArrayOfObject} result from db.
 */
export const selectById = async (id) => {
  const query = {
    type: 'select',
    table: 'public.users',
    condition: { id },
    limit: 5,
  };

  const result = await seedingData(query, 'service_auth');

  return result;
};
