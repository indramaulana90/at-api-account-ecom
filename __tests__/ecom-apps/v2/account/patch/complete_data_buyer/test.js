/* eslint-disable no-unused-vars */
/**
* Author      : Evinna Ginola Utama (evinna.ginola@sinbad.co.id)
* Modifier    :
* Domain      : account
* Description : This Main File for Testing on API patch_complete_data_buyer
*/

import getDataFromPool from '@at-api-core/get-data-from-pool';
import * as er from '@at-api-core/expect-response';
import endpoint from './endpoint';
import * as data from './data';
import tc from './tc';
import schema from './schema.json';
import schemaError from '../../../../../../helper/schema/schema_error_default.json';
import rs from '../../../../../../helper/responseStatusLibrary.json';
import { resetAccountChimera as resetAuthUsers } from '../../../../../../data/model/v1/auth/users';
import { resetAccountChimera as resetAccUsers, selectById as selectUsers } from '../../../../../../data/model/v1/account/users';
import { resetAccountChimera as resetAccBuyers, selectById as selectBuyers } from '../../../../../../data/model/v1/account/buyers';

const { cookies, buyerId } = getDataFromPool('ng','sinbad-app', 'Chimera-SR_081231111111');
const userId = getDataFromPool('ng','sinbad-app', 'Chimera-SR_081231111111').user.id;

const pause = require('@at-api-core/pause');

describe(`${tc.type[1]} ${tc.describe}`, () => {
  afterAll(async () => {
    await resetAuthUsers(userId);
    await resetAccUsers(userId);
    await resetAccBuyers(buyerId);
  });
  test(`${tc.positive.default}`, async () => {
    const payload = data.update();

    resp = await endpoint(cookies, payload);
    er.expectStatus(resp, rs.statusOk);
    er.expectSchema(resp, schema);
    await pause(1000);

    const arrDBUser = await selectUsers(userId);
    const objectDBUser = (arrDBUser)[0];
    console.log("response: ", objectDBUser);
    er.expectEqualDB(resp, payload.user, objectDBUser, data.compareDBUser);
    const arrDBBuyer = await selectBuyers(buyerId);
    const objectDBBuyer = (arrDBBuyer)[0];
    er.expectEqualDB(resp, payload.buyer, objectDBBuyer, data.compareDBBuyer);
  });
});

describe(`${tc.type[0]} ${tc.describe}`, () => {
  afterAll(async () => {
    await resetAuthUsers(userId);
    await resetAccUsers(userId);
    await resetAccBuyers(buyerId);
  });
  data.negativeCaseUser.forEach(async (element) => {
    test(element.title, async () => {
      const payload = data.update();
      (payload.user)[Object.keys(element.value)] = Object.values(element.value)[0];

      resp = await endpoint(cookies, payload);
      er.expectStatus(resp, rs.statusUnprocessableEntity);
      er.expectSchema(resp, schemaError);
    });
  });
  data.negativeCaseBuyer.forEach(async (element) => {
    test(element.title, async () => {
      const payload = data.update();
      (payload.buyer)[Object.keys(element.value)] = Object.values(element.value)[0];

      resp = await endpoint(cookies, payload);
      er.expectStatus(resp, rs.statusUnprocessableEntity);
      er.expectSchema(resp, schemaError);
    });
  });
  test(tc.negative.vehAccIdNotExist, async () => {
    const payload = data.update();
    (payload.buyer).vehicleAccessibilityId = 7;

    resp = await endpoint(cookies, payload);
    er.expectStatus(resp, rs.statusBadRequest);
    er.expectSchema(resp, schemaError);
    er.expectSnapshot(resp);
  });
});

describe(`${tc.type[0]} ${tc.describe}`, () => {
  test(`${tc.negative.xPlatformInvalid}`, async () => {
    const payload = data.update();

    resp = await endpoint(cookies, payload).set('x-platform', 'error');
    er.expectStatus(resp, rs.statusUnauthorized);
    er.expectUnauthorized(resp);
  });

  test(`${tc.negative.xPlatformNotSet}`, async () => {
    const payload = data.update();

    resp = await endpoint(cookies, payload).unset('x-platform');
    er.expectStatus(resp, rs.statusUnauthorized);
    er.expectUnauthorized(resp);
  });

  data.loopingErrorCookies.forEach(async (element) => {
    test(`${element.title}`, async () => {
      const payload = data.update();
      const cookiesNegative = element.value;

      resp = await endpoint(cookiesNegative, payload);
      er.expectStatus(resp, rs.statusUnauthorized);
      er.expectUnauthorized(resp);
    });
  });
});
