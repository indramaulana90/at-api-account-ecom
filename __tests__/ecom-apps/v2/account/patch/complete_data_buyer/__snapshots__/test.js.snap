// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@negative patch complete_data_buyer -- As a Store, I should not be able to update my email if it does not match the email format 1`] = `
Object {
  "code": 40010000001,
  "errorMessage": Object {
    "email": "Email must be valid email format",
  },
  "message": "Unprocessable Entity",
  "type": "UNPROCESSABLE_ENTITY",
}
`;

exports[`@negative patch complete_data_buyer -- As a Store, I should not be able to update my id image if the type is not a string 1`] = `
Object {
  "code": 40010000001,
  "errorMessage": Object {
    "idImageUrl": "idImageUrl must be an URL address",
  },
  "message": "Unprocessable Entity",
  "type": "UNPROCESSABLE_ENTITY",
}
`;

exports[`@negative patch complete_data_buyer -- As a Store, I should not be able to update my idNo if less than 16 char 1`] = `
Object {
  "code": 40010000001,
  "errorMessage": Object {
    "idNo": "idNo length must be equal to 16",
  },
  "message": "Unprocessable Entity",
  "type": "UNPROCESSABLE_ENTITY",
}
`;

exports[`@negative patch complete_data_buyer -- As a Store, I should not be able to update my idNo if more than 16 char 1`] = `
Object {
  "code": 40010000001,
  "errorMessage": Object {
    "idNo": "idNo length must be equal to 16",
  },
  "message": "Unprocessable Entity",
  "type": "UNPROCESSABLE_ENTITY",
}
`;

exports[`@negative patch complete_data_buyer -- As a Store, I should not be able to update my idNo if the type is not a number string 1`] = `
Object {
  "code": 40010000001,
  "errorMessage": Object {
    "idNo": "idNo must be a number string",
  },
  "message": "Unprocessable Entity",
  "type": "UNPROCESSABLE_ENTITY",
}
`;

exports[`@negative patch complete_data_buyer -- As a Store, I should not be able to update my latitude if the type is a string 1`] = `
Object {
  "code": 40010000001,
  "errorMessage": Object {
    "latitude": "latitude must be a latitude string or number",
  },
  "message": "Unprocessable Entity",
  "type": "UNPROCESSABLE_ENTITY",
}
`;

exports[`@negative patch complete_data_buyer -- As a Store, I should not be able to update my locationId with invalid format 1`] = `
Object {
  "code": 40010000001,
  "errorMessage": Object {
    "locationId": "locationId must be a string",
  },
  "message": "Unprocessable Entity",
  "type": "UNPROCESSABLE_ENTITY",
}
`;

exports[`@negative patch complete_data_buyer -- As a Store, I should not be able to update my longitude if the type is a string 1`] = `
Object {
  "code": 40010000001,
  "errorMessage": Object {
    "longitude": "longitude must be a longitude string or number",
  },
  "message": "Unprocessable Entity",
  "type": "UNPROCESSABLE_ENTITY",
}
`;

exports[`@negative patch complete_data_buyer -- As a Store, I should not be able to update my name if the type is not a string 1`] = `
Object {
  "code": 40010000001,
  "errorMessage": Object {
    "name": "name must be a string",
  },
  "message": "Unprocessable Entity",
  "type": "UNPROCESSABLE_ENTITY",
}
`;

exports[`@negative patch complete_data_buyer -- As a Store, I should not be able to update my noteAddress if the type is not a string 1`] = `
Object {
  "code": 40010000001,
  "errorMessage": Object {
    "noteAddress": "noteAddress must be a string",
  },
  "message": "Unprocessable Entity",
  "type": "UNPROCESSABLE_ENTITY",
}
`;

exports[`@negative patch complete_data_buyer -- As a Store, I should not be able to update my selfie image if the type is not a string 1`] = `
Object {
  "code": 40010000001,
  "errorMessage": Object {
    "selfieImageUrl": "selfieImageUrl must be an URL address",
  },
  "message": "Unprocessable Entity",
  "type": "UNPROCESSABLE_ENTITY",
}
`;

exports[`@negative patch complete_data_buyer -- As a Store, I should not be able to update my store address if the type is not a string 1`] = `
Object {
  "code": 40010000001,
  "errorMessage": Object {
    "address": "address must be a string",
  },
  "message": "Unprocessable Entity",
  "type": "UNPROCESSABLE_ENTITY",
}
`;

exports[`@negative patch complete_data_buyer -- As a Store, I should not be able to update my store name if the type is not a string 1`] = `
Object {
  "code": 40010000001,
  "errorMessage": Object {
    "name": "name must be a string",
  },
  "message": "Unprocessable Entity",
  "type": "UNPROCESSABLE_ENTITY",
}
`;

exports[`@negative patch complete_data_buyer -- As a Store, I should not be able to update my store phone number if the type is not a number string 1`] = `
Object {
  "code": 40010000001,
  "errorMessage": Object {
    "phoneNo": "phoneNo must be a number string",
  },
  "message": "Unprocessable Entity",
  "type": "UNPROCESSABLE_ENTITY",
}
`;

exports[`@negative patch complete_data_buyer -- As a Store, I should not be able to update my store photo if the type is not a string 1`] = `
Object {
  "code": 40010000001,
  "errorMessage": Object {
    "imageUrl": "imageUrl must be an URL address",
  },
  "message": "Unprocessable Entity",
  "type": "UNPROCESSABLE_ENTITY",
}
`;

exports[`@negative patch complete_data_buyer -- As a Store, I should not be able to update my tax image if the type is not a string 1`] = `
Object {
  "code": 40010000001,
  "errorMessage": Object {
    "taxImageUrl": "taxImageUrl must be an URL address",
  },
  "message": "Unprocessable Entity",
  "type": "UNPROCESSABLE_ENTITY",
}
`;

exports[`@negative patch complete_data_buyer -- As a Store, I should not be able to update my taxNo if less than 15 char 1`] = `
Object {
  "code": 40010000001,
  "errorMessage": Object {
    "taxNo": "taxNo length must be equal to 15",
  },
  "message": "Unprocessable Entity",
  "type": "UNPROCESSABLE_ENTITY",
}
`;

exports[`@negative patch complete_data_buyer -- As a Store, I should not be able to update my taxNo if more than 15 char 1`] = `
Object {
  "code": 40010000001,
  "errorMessage": Object {
    "taxNo": "taxNo length must be equal to 15",
  },
  "message": "Unprocessable Entity",
  "type": "UNPROCESSABLE_ENTITY",
}
`;

exports[`@negative patch complete_data_buyer -- As a Store, I should not be able to update my taxNo if the type is not a number string 1`] = `
Object {
  "code": 40010000001,
  "errorMessage": Object {
    "taxNo": "taxNo must be a number string",
  },
  "message": "Unprocessable Entity",
  "type": "UNPROCESSABLE_ENTITY",
}
`;

exports[`@negative patch complete_data_buyer -- As a Store, I should not be able to update my vehicleAccessibilityAmount if the type is not a number 1`] = `
Object {
  "code": 40010000001,
  "errorMessage": Object {
    "vehicleAccessibilityAmount": "vehicleAccessibilityAmount must be a number conforming to the specified constraints",
  },
  "message": "Unprocessable Entity",
  "type": "UNPROCESSABLE_ENTITY",
}
`;

exports[`@negative patch complete_data_buyer -- As a Store, I should not be able to update my vehicleAccessibilityAmount if the value more than 4 1`] = `
Object {
  "code": 40010000001,
  "errorMessage": Object {
    "vehicleAccessibilityAmount": "vehicleAccessibilityAmount must not be greater than 2",
  },
  "message": "Unprocessable Entity",
  "type": "UNPROCESSABLE_ENTITY",
}
`;

exports[`@negative patch complete_data_buyer -- As a Store, I should not be able to update my vehicleAccessibilityId if the number does not exist 1`] = `
Object {
  "code": 40020000029,
  "errorMessage": "ID does not exist in table vehicle_accessibilities.",
  "message": "Invalid vehicleAccessibilityId",
  "type": "BAD_REQUEST",
}
`;

exports[`@negative patch complete_data_buyer -- As a Store, I should not be able to update my vehicleAccessibilityId if the type is not a number 1`] = `
Object {
  "code": 40010000001,
  "errorMessage": Object {
    "vehicleAccessibilityId": "vehicleAccessibilityId must be a number conforming to the specified constraints",
  },
  "message": "Unprocessable Entity",
  "type": "UNPROCESSABLE_ENTITY",
}
`;
